tasks/ubuntu/bionic.task/os_boot.erb

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+exec >> /var/log/razor.log 2>&1
+
+# ubuntu 18.04 networking doesn't wake up right away, so...
+for i in {1..15}; do
+  curl -s <%= compose_url %> >/dev/null && break
+  sleep 1
+done
+
+<%= render_template("set_hostname") %>
+
+# APT configuration and system update
+sed -i s$'\001'<%= repo_url %>$'\001'http://gb.archive.ubuntu.com/ubuntu$'\001' /etc/apt/sources.list
+
+apt-get -y update
+[ "$?" -eq 0 ] || curl -s <%= log_url("apt_update_fail", :error) %>
+
+apt-get -y upgrade
+[ "$?" -eq 0 ] || curl -s <%= log_url("apt_upgrade_fail", :error) %>
+
+apt-get -y dist-upgrade
+[ "$?" -eq 0 ] || curl -s <%= log_url("apt_dist_upgrade_fail", :error) %>
+
+
+<%= render_template("store_ip") %>
+
+<%= render_template("os_complete") %>
+
+# Send final state
+curl -s <%= stage_done_url("os_boot") %>
```

Comment:
the one inherited is sending the finished message not specifying the state.

is "finished" ok in this case?

Reply:
Specifying the state, you mean stating that the task is finished?